apps/src/code-studio/components/WorkspaceAlert.jsx

```diff
@@ -0,0 +1,54 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Alert from '@cdo/apps/templates/alert';
+import $ from 'jquery';
+
+export default class WorkspaceAlert extends React.Component {
+  static propTypes = {
+    type: PropTypes.oneOf(['error', 'warning', 'notification']).isRequired,
+    children: PropTypes.element,
+    onClose: PropTypes.func,
+    isBlockly: PropTypes.bool,
+    isCraft: PropTypes.bool,
+    displayBottom: PropTypes.bool
+  };
+
+  render() {
+    const {displayBottom} = this.props;
+    var toolbarWidth;
+    if (this.props.isBlockly && this.props.isCraft) {
+      // craft has a slightly different way of constructing the toolbox so we need to use
+      // the toolbox header's width to get the width of the actual toolbox.
+      toolbarWidth = $('#toolbox-header').width();
+    } else if (this.props.isBlockly) {
+      toolbarWidth = $('.blocklyToolboxDiv').width();
+    } else {
+      toolbarWidth =
+        $('.droplet-palette-element').width() + $('.droplet-gutter').width();
+    }
+
+    let style = {
+      position: 'absolute',
+      right: 0,
+      left: toolbarWidth
+    };
```

Comment:
Our convention has generally been to put the styles in a const object outside the class (i.e. line 5). I'm actually not completely sure why or if that's best practice, so definitely open to other ways or a second opinion. If you want to do this, it would go like this:
```
const styles = {
  alertContainer: {
    position: 'absolute',
    right: 0,
    ---
  }
}

export default class WorkspaceAlert extends React.Component {
  ---
  render(
    ---
    const position = displayBottom ? {bottom: 0} : {top: $('#headers').height()}
    return (
      <div style={{...styles.container, ...position}}>
      ---
```